### ISSUE TITLE
Warnings about illegal reflective access / InvocationTargetException

### ISSUE THREAD

**Issue by 3r1d** [nextflow-io/nextflow#3663]
### Expected behavior and actual behavior

Expected behaviour => no warning.
Actual behaviour => warnings.

### Steps to reproduce the problem

1. create this files tree

```
mkdir -p /tmp/testdata/l1/l2
echo 1 > /tmp/testdata/l1/l2/f1
echo 1 > /tmp/testdata/l1/l2/f2

```

2. run the workflow below

```
nextflow.enable.dsl=2

process myprocess {
    input:
        val a
    output:
        val "foo"
    script:
        m = a =~ /^\/\w+\/(\w+)/
        """
        """
}

workflow {
    channel.fromPath("/tmp/testdata/*/*/*") | myprocess
}

```

### Program output 

with java 11

```
WARNING: An illegal reflective access operation has occurred

WARNING: Illegal reflective access by
com.esotericsoftware.kryo.serializers.FieldSerializer
(file:/home/jerome/.nextflow/capsule/deps/com/esotericsoftware/kryo/kryo/2.24.0/kryo-2.24.0.jar)
to field java.util.regex.Matcher.parentPattern

WARNING: Please consider reporting this to the maintainers of
com.esotericsoftware.kryo.serializers.FieldSerializer

```
with java 17

```
WARN: Cannot serialize context map. Cause:
java.lang.reflect.InvocationTargetException -- Resume will not work on this
process

```

### Environment 

* Nextflow version: version 22.10.6 build 5843
* Java version: 11 and 17
* Operating system: ubuntu 20.04
* Bash version: 5.0.17

**Comment by bentsherman** [comment]
Can you share the nextflow logs for these runs? Need to see the stack trace.

**Comment by crcmt** [comment]
[nextflow.log](https://github.com/nextflow-io/nextflow/files/10794979/nextflow.log)

**Comment by crcmt** [comment]
[nextflow.log](https://github.com/nextflow-io/nextflow/files/10795002/nextflow.log)

**Comment by 3r1d** [comment]
[nextflow.log](https://github.com/nextflow-io/nextflow/files/10795012/nextflow.log)

**Comment by bentsherman** [comment]
Looks like the kryo serializer doesn't support `Matcher` objects. I suspect this issue would be fixed by updating kryo to a newer version.

See this branch: https://github.com/nextflow-io/nextflow/tree/ben-update-kryo

You're welcome to test it if you like. We have some other issues related to kryo, but we're waiting for a good time to update it because it will break everyone's resume cache.

**Comment by pditommaso** [comment]
it makes no sense to serialise the `Matcher`! it's enough to add `def` in front of the declaration eg.

```
def m = a =~ /^\/\w+\/(\w+)/
```

**Comment by pditommaso** [comment]
Fixed 6aecc4ed
